### ISSUE TITLE
Colour Picker Broken Lua Error

### ISSUE THREAD

**Issue by Nukesoup** [kesava-wow/kuinameplates2#704]
**Colour Picker broken and presenting an Lua error for all colour options in Kui Nameplates Core.**

This issue only came about since 10.2.5 patch and is not affecting Kui Target Helper.

Lua Error presented when selecting any Colour Picker option in Kui Nameplates Core:

Message: ...erface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:421: attempt to call method 'SetColorRGB' (a nil value)
Time: Sun Feb 11 20:23:58 2024
Count: 2
Stack: ...erface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:421: attempt to call method 'SetColorRGB' (a nil value)
[string "@Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua"]:421: in function <...erface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:410>

Locals: self = KuiNameplatesCoreConfigtarget_glow_colourColourPicker {
 0 = <userdata>
 enabled = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/create.lua:70
 label = FontString {
 }
 env = "target_glow_colour"
 block = Frame {
 }
 Set = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:401
 Get = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:396
}
val = <table> {
 1 = 0.815686
 2 = 0
 3 = 1
 4 = 1
}
(*temporary) = nil
(*temporary) = ColorPickerFrame {
 0 = <userdata>
 GetColorRGB = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:79
 DragBar = Frame {
 }
 OnKeyDown = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:53
 cancelFunc = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:383
 previousValues = <table> {
 }
 SetupColorPickerAndShow = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:62
 OnLoad = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:3
 opacity = 0
 OnShow = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:34
 GetPreviousValues = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:91
 GetColorAlpha = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:83
 hasOpacity = true
 GetExtraInfo = <function> defined @Interface/FrameXML/ColorPickerFrame.lua:87
 Header = Frame {
 }
 Footer = Frame {
 }
 func = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:383
 Border = Frame {
 }
 Content = Frame {
 }
 opacityFunc = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:383
}
(*temporary) = 0.815686
(*temporary) = 0
(*temporary) = 1
(*temporary) = "attempt to call method 'SetColorRGB' (a nil value)"
opt = KuiNameplatesCoreConfig {
 0 = <userdata>
 LayoutLoaded = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/main.lua:575
 CreateCheckBox = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:116
 CreatePopupPage = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:546
 Locale = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/main.lua:513
 TabList = KuiNameplatesCoreConfigTabList {
 }
 CreateConfigPage = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:529
 CreateButton = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:71
 pages = <table> {
 }
 ScrollFrame = KuiNameplatesCoreConfigPageScrollFrame {
 }
 CurrentPage_UpdateClipboardButton = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:1064
 PageBG = Frame {
 }
 GetLocale = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/main.lua:542
 ConfigChanged = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/main.lua:547
 profile = <table> {
 }
 name = "Kui Nameplates Core"
 LocaleLoaded = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/main.lua:523
 ClipboardButton = Button {
 }
 CurrentPage_Copy = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:1004
 CurrentPage_Name = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:999
 Initialise = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:1161
 CurrentPage_ClipboardButtonClick = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:1048
 CreateSlider = <function> defined @Interface/AddOns/Kui_Nameplates_Core_Config/helpers.lua:314
 CurrentPage_Paste = <function> defined @Interface/AddOns/Ku

![Capture](https://github.com/kesava-wow/kuinameplates2/assets/159715756/80d8d4aa-df9c-4cb5-9a87-e65593df3a33)

**Comment by BartekBanas** [comment]
For me, the color picker doesn't open at all

**Comment by Nukesoup** [comment]
> For me, the color picker doesn't open at all

Yeah sorry if I didn't make it clear in my original post, mine won't open at all either, just presents a Lua error as soon as I try to use anything that uses the current colour picker.
